app/services/developer_service.rb

```diff
@@ -6,9 +6,21 @@ def initialize(params)
 
   def register_as_developer
     if @user && !@user.developer?
-      ApiKey.create(user: @user)
-      @user.roles << Role.find_or_create_by(name: 'developer')
+      key = generate_key
```

Comment:
@jmejia controller didn't have to be changed, it now raises an exception if the validation fails for whatever reason that can be caught elsewhere

Reply:
Much better. Nice 👍